File: trunk/openrest4net/openrest4net-api/com/googlecode/openrest/OrderConfirmation.cs
using System;

namespace com.googlecode.openrest
{
    class OrderConfirmation
    {
        public OrderConfirmation(Order order, String message)
        {
            this.order = order;
            this.message = message;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        internal OrderConfirmation() { }

        /** The confirmed order. */
        public Order order;

        /** The restaurant's confirmation message. */
        public string message;
    }
}

File: trunk/openrest4j/openrest4j-api/src/com/googlecode/openrest/v1_0/OrderConfirmation.java
package com.googlecode.openrest.v1_0;

import java.io.Serializable;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class OrderConfirmation implements Serializable {
    public OrderConfirmation(Order order, String message) {
        this.order = order;
        this.message = message;
    }
    
    /** Default constructor for JSON deserialization. */
    public OrderConfirmation() {}

    /** The confirmed order. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public Order order;

    /** The restaurant's confirmation message. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String message;
    
    private static final long serialVersionUID = 1L;
}
